util/logs/workflow-logger.go

```diff
@@ -111,15 +132,24 @@ func WorkflowLogs(ctx context.Context, wfClient versioned.Interface, kubeClient
 					logCtx.Error(err)
 					return
 				}
+
 				scanner := bufio.NewScanner(stream)
+				//give it more space for long line
+				scanner.Buffer(make([]byte, startBufSize), maxTokenLength)
+				//avoid bufio.ErrTooLong error when encounters a very very long line
+				scanner.Split(scanLinesOrGiveLong)
 				for scanner.Scan() {
 					select {
 					case <-ctx.Done():
 						return
 					default:
 						line := scanner.Text()
 						parts := strings.SplitN(line, " ", 2)
-						content := parts[1]
+						//on old version k8s, the line may contains no space, hence len(parts) would equal to 1
+						content := ""
```

Comment:
How old is your k8s cluster?

Reply:
1.18

Reply:
Ok, I see.
But, in my opinion, it may still be necessary to ensure that the length of parts is greater than 1 here.
If there is a very very long token, which doesn't contain any space in it, the length of parts would equal to 1, thus it
 would cause a crash.

Reply:
This sounds like additional scan that could be time-consuming when there are a lot of logs

Reply:
Actually, there isn't additional scan here. `scanLinesOrGiveLong` is just a custom scan function replacing the default one. Comparing the default scan which is `bufio.ScanLines`, `scanLinesOrGiveLong` just a simple function wrapping `bufio.ScanLines` with some extra checking to ensure the token returned wouldn't be too large to cause `bufio.ErrTooLong` error.

Reply:
@alexec Any objections?